gluecodium/src/main/resources/templates/jni/utils/JniCallbackErrorCheckingImplementation.mustache

```diff
@@ -0,0 +1,87 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2023 HERE Europe B.V.
+  !
+  ! Licensed under the Apache License, Version 2.0 (the "License");
+  ! you may not use this file except in compliance with the License.
+  ! You may obtain a copy of the License at
+  !
+  !     http://www.apache.org/licenses/LICENSE-2.0
+  !
+  ! Unless required by applicable law or agreed to in writing, software
+  ! distributed under the License is distributed on an "AS IS" BASIS,
+  ! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ! See the License for the specific language governing permissions and
+  ! limitations under the License.
+  !
+  ! SPDX-License-Identifier: Apache-2.0
+  ! License-Filename: LICENSE
+  !
+  !}}
+{{>java/CopyrightHeader}}
+
+#include "JniCallbackErrorChecking.h"
+
+#include "JniCallJavaMethod.h"
+#include "JniCppConversionUtils.h"
+
+#include <string>
+
+{{#internalNamespace}}
+namespace {{.}}
+{
+{{/internalNamespace}}
+namespace jni
+{
+
+void checkExceptionAndReportIfAny(JNIEnv *env) {
+
+    if (! env->ExceptionCheck( ) )
+    {
+        return;
+    }
+
+    jthrowable exception = env->ExceptionOccurred();
+    env->ExceptionClear( );
+
+    std::string error_msg;
+
+    auto frames = call_java_method<jobjectArray>(env, exception, "getStackTrace", "()[Ljava/lang/StackTraceElement;");
+    auto framesLength = static_cast< size_t >( env->GetArrayLength( frames.get() ) );
+
+    if (0 == frames || 0 == framesLength) {
+        auto exceptionClass = get_object_class(env, exception);
+        auto exceptionClassName = call_java_method<jstring>(env, exceptionClass, "getName", "()Ljava/lang/String;");
+        auto exceptionClassNameString = convert_from_jni(env , exceptionClassName, (std::string*)nullptr);
+        env->FatalError(exceptionClassNameString.c_str());
+    }
+
+    auto exceptionToString = make_local_ref(env, call_java_method<jstring>(env, exception, "toString", "()Ljava/lang/String;").get());
+    std::string actualExceptionMessage = convert_from_jni(env , exceptionToString, (std::string*)nullptr);
+    error_msg += actualExceptionMessage;
+
+    jsize i = 0;
+    for (i = 0; i < framesLength; i++)
```

Comment:
I recalled we have a function to convert array in ArrayConversionUtils.h, try it:
`auto framesVector = convert_from_jni(env, frames, (std::vector< std::string>*) nullptr);`
May be it makes sense to call it early at line 50

Reply:
Just save `frames` as reference to `jobject` instead of `jobjectArray`:

```
auto frames = call_java_method<jobject>(env, exception, "getStackTrace", "()[Ljava/lang/StackTraceElement;");
```